frontend/src/app/api/user/update-tokens.ts

```diff
@@ -0,0 +1,56 @@
+import { createClient } from "@supabase/supabase-js";
+import { type NextRequest } from "next/server";
+
+export async function POST(request: NextRequest) {
+  const webhookSecret = request.headers.get('x-webhook-secret');
+  if (webhookSecret !== process.env.WEBHOOK_SECRET) {
+    return new Response(JSON.stringify({ error: 'Unauthorized' }), {
+      status: 403,
+      headers: { 'Content-Type': 'application/json' }
+    });
+  }
+
+  const supabase = createClient(
+    process.env.DB_URL as string,
+    process.env.DB_SECRET_KEY as string
+  );
+
+  const payload = await request.json() as any;
```

Comment:
```suggestion
  const payload: DeletePayload<CurrentMatchData> = await request.json();
```

then declare this type at top of file [[ref](https://supabase.com/docs/guides/database/webhooks#payload)]:
```ts
import { CurrentMatchData } from "@/schemas";

type DeletePayload<T = object> = {
  type: 'DELETE';
  table: string;
  schema: string;
  record: null;
  old_record: T;
};
```